sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -651,41 +677,58 @@ abstract class CodeGenerator[InType <: AnyRef, OutType <: AnyRef] extends Loggin
       sys.error(s"Code generation not support for hashset of type $unsupportedType")
   }
 
-  protected def primitiveForType(dt: DataType) = dt match {
-    case IntegerType => "Int"
+  protected def primitiveForType(dt: DataType): String = dt match {
+    case IntegerType => "int"
+    case LongType => "long"
+    case ShortType => "short"
+    case ByteType => "byte"
+    case DoubleType => "double"
+    case FloatType => "float"
+    case BooleanType => "boolean"
+    case dt @ DecimalType() => "org.apache.spark.sql.types.Decimal"
+    case BinaryType => "byte[]"
+    case StringType => "org.apache.spark.sql.types.UTF8String"
```

Comment:
We could do `classOf[UTF8String].getName` here instead.  This would save us some work if we relocate `UTF8String` to the `unsafe` package in the future.